src/test/java/hudson/tasks/junit/JUnitResultArchiverTest.java

```diff
@@ -96,20 +100,27 @@
 public class JUnitResultArchiverTest {
 
     @Rule public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public LoggerRule logRule = new LoggerRule().recordPackage(JUnitResultArchiver.class, Level.FINE);
+
+    @ClassRule
+    public final static BuildWatcher buildWatcher = new BuildWatcher();
+
     private FreeStyleProject project;
     private JUnitResultArchiver archiver;
 
     @Before public void setUp() throws Exception {
         project = j.createFreeStyleProject("junit");
         archiver = new JUnitResultArchiver("*.xml");
         project.getPublishersList().add(archiver);
-
         project.getBuildersList().add(new TouchBuilder());
     }
 
     @LocalData("All")
     @Test public void basic() throws Exception {
-        FreeStyleBuild build = project.scheduleBuild2(0).get(10, TimeUnit.SECONDS);
+        FreeStyleBuild build =
+                j.assertBuildStatus(Result.UNSTABLE, j.waitForCompletion(project.scheduleBuild2(0).waitForStart()));
```

Comment:
```suggestion
        FreeStyleBuild build = j.buildAndAssertStatus(Result.UNSTABLE, project);
```

As in https://github.com/jenkinsci/junit-plugin/pull/384#discussion_r874749729 it would be better to file such cleanups as a separate `chore` / `tests` PR that explicitly makes no change to behavior and could be merged earlier (or later) to avoid confusing and delaying the review of this PR.